InnerEye/ML/augmentations/transform_pipeline.py

```diff
@@ -0,0 +1,168 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from typing import Any, Callable, List, Union
+
+import PIL
+import torch
+
+from torchvision.transforms import CenterCrop, ColorJitter, Compose, RandomAffine, RandomErasing, \
+    RandomHorizontalFlip, RandomResizedCrop, Resize
+from torchvision.transforms.functional import to_tensor
+from yacs.config import CfgNode
+
+from InnerEye.ML.augmentations.image_transforms import AddGaussianNoise, ElasticTransform, ExpandChannels, RandomGamma
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from InnerEye.ML.dataset.scalar_sample import ScalarItem
+
+
+class ImageTransformationPipeline:
+    """
+    This class is the base class to classes built to define data augmentation transformations
+    for 3D or 2D image inputs.
```

Comment:
This actually works for images to. In the caller here, if the input is scalar item it applies it to the relevant attributes (img or seg) otherwise it applies it to the tensor directly. But maybe indeed we can rethink whether we can instead simply ask the user to returns a pipeline for images and one for segmentations and the data loading code makes the conversion. Will rethink it

Reply:
I now changed the worflow: the image pipeline now really just takes care of create a pipeline of transforms that works for 2D or 3D images (tensor or PIL.Image inputs). The conversion to ScalarItem transformation is now delegated to scalar_config. I also disantangled the transforms for image and segmentations map: we now have two functions in the scalar_config API one is "get_image_transform" that is used to set transforms to apply to images and "get_segmentations_transform" that is used to set transforms to apply to segmentations. That way for the 99,999% of cases where we only uses images input we don't need to worry about segmentation maps transforms.